src/main/java/logic/Node.java

```diff
@@ -1,27 +1,98 @@
+    public static Node STUB = new Node("stub");
 
-    private NetworkLogic networkLogic;
-    private SellerLogic sellerLogic;
+    public final String name;
 
-    void sendMessage(String node, Message message) {
-        //networkLogic.send(node, message);
-        System.out.println("Test: " + node + "->" + message);
+    private final NetworkLogic networkLogic = new NetworkLogic();
+    private ParticipantState currentState = initState();
+    private final SellerLogic sellerLogic = new SellerLogic(this, currentState.getDocuments());
+
+    private final BlockingQueue<AddressedMessage> messagesToSend = new LinkedBlockingDeque<>();
+    private final BlockingQueue<AddressedMessage> messagesToHandle = new LinkedBlockingDeque<>();
+
+    private final Thread sendingThread;
+    private final Thread handlingThread;
+
+    public Node(String name) {
+        this.name = name;
+
+        networkLogic.addMessageHandler(messagesToHandle::add);
+
+        sendingThread = new Thread(this::sendMessagesLoop);
+        sendingThread.run();
+        handlingThread = new Thread(this::handleMessagesLoop);
+        handlingThread.run();
     }
 
-    /**
-     *
-     * @return A unique name of this node
-     */
     public String getName() {
-        return "SellerTestNode";
-        //return this.toString();
+        return name;
+    }
+
+    private ParticipantState initState() {
+        //todo read from file or console or whatever
+        return new ParticipantState(
+                1000,
+                Arrays.asList(File.parseFile("a=123"), File.parseFile("b=456")),
+                Collections.emptySet(),
+                Collections.emptyMap());
+    }
+
+    public void shutdown() {
+        sendingThread.interrupt();
+        handlingThread.interrupt();
+    }
+
+    void sendMessage(String node, Message message) {
+        System.out.println("message to send: " + node + "->" + message);
+        messagesToSend.add(new AddressedMessage(node, message));
+    }
+
+    // Run in a separate thread
+    private void sendMessagesLoop() {
+        while (!Thread.interrupted()) {
+            try {
+                AddressedMessage m = messagesToSend.take();
+                networkLogic.send(m.node, m.message);
+            } catch (InterruptedException e) {
+                break;
+            }
+        }
+    }
+
+    private void handleMessage(String node, Message message) {
+        System.out.println("message to send: " + node + "->" + message);
+        //todo forward the message to sellerLogic, update currentState etc.
+    }
+
+    // Run in a separate thread
+    private void handleMessagesLoop() {
+        while (!Thread.interrupted()) {
+            try {
+                AddressedMessage m = messagesToHandle.take();
+                handleMessage(m.node, m.message);
+            } catch (InterruptedException e) {
+                break;
+            }
+        }
     }
 }
+
+class AddressedMessage {
```

Comment:
Why you need separate class for this? Use existing one please. 
(https://github.com/21011996/Stock-exchange/tree/master/src/main/java/messages)